src/RingGrids/quadrature_weights.jl

```diff
@@ -3,19 +3,26 @@
 # clenshaw_curtis_weights are exact for equi-angle latitudes when nlat > 2T+1
 gaussian_weights(nlat_half::Integer) = FastGaussQuadrature.gausslegendre(2nlat_half)[2][1:nlat_half]
 function clenshaw_curtis_weights(nlat_half::Integer)
-    nlat = 2nlat_half - 1
+    nlat = get_nlat(FullClenshawArray, nlat_half)
     θs = get_colat(FullClenshawGrid, nlat_half)
     return [4sin(θj)/(nlat+1)*sum([sin(p*θj)/p for p in 1:2:nlat]) for θj in θs[1:nlat_half]]
 end
 
 # QUADRATURE WEIGHTS (INEXACT), for HEALPix full grids
-healpix_weights(nlat_half::Integer) =
-    [nlon_healpix(nlat_half, j) for j in 1:nlat_half].*(2/npoints_healpix(nlat_half))
-octahealpix_weights(nlat_half::Integer) =
-    [nlon_octahealpix(nlat_half, j) for j in 1:nlat_half].*(2/npoints_octahealpix(nlat_half))
+function equal_area_weights(Grid::Type{<:AbstractGridArray}, nlat_half::Integer)
+    weights = zeros(nlat_half)
+    for j in 1:nlat_half
+        nlon = get_nlon_per_ring(Grid, nlat_half, j)
+        weights[j] = 2nlon/get_npoints2D(Grid, nlat_half)
+    end
+    return weights
```

Comment:
I assume the quadrature weights need to go on the GPU as well when the Grid is on the GPU. But I would say that we can also adjust that when we tackle the transform submodule

Reply:
Yes, they're always precomputed in stored in `SpectralTransform`, (and there's more vectors in there that need to be array-agnostic) so I'd suggest we don't change this here but in `SpeedyTransforms`